CMakeLists.txt

```diff
@@ -273,22 +273,26 @@ target_compile_options(freertos_kernel PRIVATE
 
 
 ########################################################################
-add_subdirectory(include)
-add_subdirectory(portable)
+include(${CMAKE_CURRENT_LIST_DIR}/include/CMakeLists.txt)
+include(${CMAKE_CURRENT_LIST_DIR}/portable/CMakeLists.txt)
```

Comment:
Can you explain why the usage of `CMAKE_CURRENT_LIST_DIR` is an improper use of CMAKE?

The idea with this change is you no longer need to include the FreeRTOS-Kernel as a sub-directory, it can then be placed anywhere in your project as all of our cmake files aren't position dependent then. So instead of now where the kernel must be included in a sub-directory of your CMake File you would instead be able to include it from anywhere in your project

Reply:
It's not that it is wrong, it is superfluous (not necessary) and is not required for what you were trying to achieve (the idea with this change).

In any cmake project, one can `add_subdirectory(free_rtos_kernel_dir)` with the appropriate configuration beforehand and that `free_rtos_kernel_dir` can be anywhere in your project.

In recent changes - the CMakeFiles.txt of the various FreeRTOS components were updated to allow a very common way of importing libraries via CMake itself:  `FetchContent`

```cmake
FetchContent_Declare( freertos_kernel
  GIT_REPOSITORY https://github.com/FreeRTOS/FreeRTOS-Kernel.git
  GIT_TAG        V10.6.1
)

...

set(FREERTOS_HEAP "4" CACHE STRING "" FORCE)
set(FREERTOS_PORT "GCC_POSIX" CACHE STRING "" FORCE)

FetchContent_MakeAvailable(freertos_kernel)
```

This will download the V10.6.1 version of the Freertos-Kernel into `build/_deps`
And then run the CMake from a local build directory in `build/_deps`.

To change the version - you then only change one line in the CMakeLists.txt that has the above example.

A great resource is [Professional CMake](https://crascit.com/professional-cmake/) for how to properly use the tool.